From c25991b89a27a7214e0984310defcc03d4af33dd Mon Sep 17 00:00:00 2001
From: Varun Sharma <vsharma@habana.ai>
Date: Wed, 13 Sep 2023 11:48:45 +0300
Subject: [PATCH] [SW-66365] gaudi3: enable signature validation check

This patch enable image signature validation check and based
on that, f/w will check for image(preboot) authenticity.

Change-Id: Ieb06e59321bccdf92b9532ff7db8aa997e664365
Signed-off-by: Varun Sharma <vsharma@habana.ai>
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 481e545..a0b2c2e 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -37,6 +37,8 @@
 #include <gaudi3_blocks_linux_driver.h>
 #include <gaudi3_reg_map.h>
 #include <pcie_dbi_masks.h>
+#include <arcboot_export.h>
+#include <hl_sec_if.h>
 
 #include <gaudi3/gaudi3_boot_info.h>
 #include <gaudi3/gaudi3_reg_map.h>
@@ -730,6 +732,15 @@
 	return num_crc;
 }
 
+static bool ppboot_sys_cfg_efuse_is_set(u32 mask)
+{
+	struct hl_fw_sec_info hl_sec_info;
+
+	arcboot_rom_get_sec_data(&hl_sec_info);
+
+	return !!(hl_sec_info.efuse_sys_cfg & mask);
+}
+
 __weak bool hl_dev_is_valid_img_desc(struct boot_desc *desc)
 {
 	u32 addr, actual_size;
@@ -739,6 +750,9 @@
 	if (hl_is_sim_mode())
 		return true;
 
+	if (!ppboot_sys_cfg_efuse_is_set(EFUSE_SEC_SIG_VAL_EN_MASK))
+		return true;
+
 	num_crc = ppboot_get_num_crc(desc->img_size);
 	actual_size = desc->img_size - (num_crc * CRC32_SIZE);
 	addr = ppboot_get_arc_addr(desc->loadaddr);
